src/Microsoft.OData.Edm/ExtensionMethods/AnnotationsTargetHelper.cs

```diff
@@ -0,0 +1,205 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="AnnotationsTargetHelper.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Edm
+{
+    using System.Collections.Generic;
+    using Microsoft.OData.Edm.Vocabularies;
+
+    /// <summary>
+    /// AnnotationsTarget helper
+    /// </summary>
+    internal static class AnnotationsTargetHelper
+    {
+        public static IEdmVocabularyAnnotatable HandleTargetSegments(IEdmModel model, List<IEdmElement> results, string[] targetSegments, int targetSegmentsCount, int level)

```

Comment:
It seems you can refine this helper methods. for example, you can use a loop within the first method to handle the non-first segments/